src/Service/JsonSchemaValidator/DataUrlConstraint.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atoolo\Form\Service\JsonSchemaValidator;
+
+use Atoolo\Form\Service\DataUrlParser;
+use finfo;
+use Opis\JsonSchema\Errors\CustomError;
+use stdClass;
+
+class DataUrlConstraint implements FormatConstraint
+{
+    public function __construct(
+        private readonly DataUrlParser $dataUrlParser,
+    ) {}
+
+    public function getType(): string
+    {
+        return 'string';
+    }
+
+    public function getName(): string
+    {
+        return 'data-url';
+    }
+
+    /**
+     * @throws CustomError
+     */
+    public function check(mixed $value, stdClass $schema): bool
+    {
+        if (!is_string($value)) {
+            throw new CustomError('Value is not a string');
+        }
```

Comment:
shouldn't this be something that is cought by the `JsonSchemaValidator`? I mean we don't also check weather or not a field expecting a number actually is one right? so why do we need to check if a `data-url` is a `string`? or could this constraint technically be configured to _any_ field? if so, is that not validated either? does this have to err at runtime?